dbeam-core/src/main/java/com/spotify/dbeam/args/QueryBuilderArgs.java

```diff
@@ -98,23 +103,44 @@ private static Boolean checkTableName(String tableName) {
 
   public static QueryBuilderArgs create(String tableName) {
     checkArgument(tableName != null,
-        "TableName cannot be null");
-    checkArgument(checkTableName(tableName),
-        "'table' must follow [a-zA-Z_][a-zA-Z0-9_]*");
+            "TableName cannot be null");
+    DbeamQueryBuilder baseSqlQuery = getBaseSqlQuery(tableName, Optional.empty());
+    return new AutoValue_QueryBuilderArgs.Builder()
+            .setTableName(tableName)
+            .setBaseSqlQuery(baseSqlQuery)
+            .setPartitionPeriod(Days.ONE)
+            .build();
+  }
+
+  public static QueryBuilderArgs create(String tableName, final Optional<String> sqlQueryOpt) {
+    checkArgument(tableName != null,
+            "TableName cannot be null");
+    DbeamQueryBuilder baseSqlQuery = getBaseSqlQuery(tableName, sqlQueryOpt);
     return new AutoValue_QueryBuilderArgs.Builder()
         .setTableName(tableName)
+        .setBaseSqlQuery(baseSqlQuery)
         .setPartitionPeriod(Days.ONE)
         .build();
   }
 
+  private static DbeamQueryBuilder getBaseSqlQuery(String tableName, Optional<String> sqlQueryOpt) {
+    checkArgument(checkTableName(tableName), "'table' must follow [a-zA-Z_][a-zA-Z0-9_]*");
+    if (!sqlQueryOpt.isPresent()) {
+      return DbeamQueryBuilder.fromTablename(tableName);
+    } else {
+      String sqlQuery = sqlQueryOpt.get();
+      return DbeamQueryBuilder.fromSqlQuery(sqlQuery);
+    }
+  }
+
   /**
    * Create queries to be executed for the export job.
    *
    * @param connection A connection which is used to determine limits for parallel queries.
    * @return A list of queries to be executed.
    * @throws SQLException when it fails to find out limits for splits.
    */
-  public Iterable<String> buildQueries(Connection connection)
+  public Iterable<DbeamQueryBuilder> buildQueries(Connection connection)
```

Comment:
Lets work on the naming a bit here. Do we need to return an `Iterable` of Builders `Iterable<DbeamQueryBuilder>` or if the caller calls build on everything, we might just return the built queries like the name of the function `buildQueries` suggests.

Reply:
Agree.